### DISCUSSION TITLE
ESPAltherma in ESPHome (was Using a olimex esp32 POE wired lan)

### DISCUSSION THREAD

**Discussion by trantoriana** [raomin/ESPAltherma#216] [Ideas]
Hi,

I have a regular esp32 with wifi connected to my altherma which works, however wifi in the basement is not ideal and sometimes has problems connecting making automations in ha based on data provided by espaltherma unreliable due to said wifi challenges.

Instead of wifi i would prefer a lan connection as provided by olimex esp32 poe.
Any change of getting espaltherma running on such a device?

Also, would it be possible to implement the serial communication and altherma protocol as esphome module? That way it would be easy enough to create your own device (using olimex poe for instance) and have native home assistant support instead of mqtt?

Thanks, trant

**Comment by raomin** [comment]
Hi @trantoriana, for the ESP32 POE, I don't see why it would not work. Just need to replace the wifi layer with ethernet.
For ESP Home, I guess it could be possible. The problem I see would be the selection of the definition file that we should either port to Home Assistant (eg inside a configuration yaml file to be pushed to the esp) or have all definition files added to the esp but I guess ESPHome would not like the footprint...

This is something we can discuss. I'll move this issue to the discussion forum as it would be more appropriate.

**Reply by berto0** [reply]
Hello, I'm trying to use an Olimex Esp32 POE board for the Espaltherma. I have little programming knowledge, so I tried ChatGPT and Copilot, but it didn't work. All I can do is see that the board is online in Home Assistant and update via OTA, but it doesn't send data via MQTT. Could you help me? I have the TX and RX pinned correctly. Thanks.

**Comment by engelchrisi** [comment]
You can try my esphome integration based on the ESPAltherma code at your own risk of course. You need some experience with custom components in esphome to do so:

https://github.com/engelchrisi/ESPAltherma
